pyrene/src/utils/DateUtils.ts

```diff
@@ -0,0 +1,258 @@
+import isExists from 'date-fns/isExists';
+import sub from 'date-fns/sub';
+import add from 'date-fns/add';
+import format from 'date-fns/format';
+import parse from 'date-fns/parse';
+import Duration from 'date-fns';
+
+import {
+  zonedTimeToUtc, utcToZonedTime,
+} from 'date-fns-tz';
+
+export enum DateUnits {
+  DAY = 'day',
+  MONTH = 'month',
+  YEAR = 'year',
+}
+
+export interface DateValidationObject {
+  dateString?: string,
+  isDateInvalid?: boolean,
+  isTimeInvalid?: boolean,
+  minimumValue?: number,
+  maximumValue?: number,
+  timeZone?: string,
+  dateFormat?: string,
+  timeString?: string,
+  timeFormat?: string,
+}
+
+export type DateType = {
+  day: number,
+  month: number,
+  year: number,
+};
+
+export type TimeType = {
+  minutes: number,
+  hours: number,
+};
+
+export const allowedValueCheck = (valueToCheck:string) : boolean => (/^[0-9.: APM]*$/.test(valueToCheck));
+
+/**
+ * Converts our custom date object to JavaScript Date
+ * Because the month of the internal object is 0-indexed and
+ * externally the dates are passed in as 1-indexed, we need to convert them
+ *
+ * @param {DateType} value
+ * @returns {Date}
+ */
+export const convertToJsDate = (date?: DateType, time?:TimeType): Date | undefined => {
+  if (date) {
+    if (time) {
+      return new Date(date.year, date.month - 1, date.day, time.hours, time.minutes);
+    }
+    return new Date(date.year, date.month - 1, date.day);
+  }
+  return undefined;
+};
+
+/**
+ * Converts a JavaScript Date object to our custom date object format
+ * Increases the month number by 1 so that it is 1-indexed
+ *
+ * @param {Date} date
+ * @returns {DateType}
+ */
+export const convertToDateTypeObject = (date: Date) : DateType => ({
```

Comment:
This type was created so that the new typescript files could work with the old JSX CalendarDateSelector component and TimeRangeSelector component and their use was for consistency. These components have changed since their typescript conversion and I guess the type is not necessary anymore.

Reply:
Function has been removed.